x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/amazonbedrock/AmazonBedrockStreamingChatProcessor.java

```diff
@@ -108,6 +111,14 @@ public void onComplete() {
         }
     }
 
+    private void runOnUtilityThreadPool(Runnable runnable) {
+        try {
+            threadPool.executor(UTILITY_THREAD_POOL_NAME).execute(runnable);
+        } catch (Exception e) {
+            logger.error(Strings.format("failed to fork [%s] to utility thread pool", runnable), e);
```

Comment:
I'll leave that for you to do in a followup. In practice we can't really throw an exception here anyway because of how the `UTILITY_THREAD_POOL_NAME` executor is configured today.